baggage/baggage.go

```diff
@@ -62,6 +63,10 @@ func NewKeyProperty(key string) (Property, error) {
 // Notice: Consider using [NewKeyValuePropertyRaw] instead
 // that does not require percent-encoding of the value.
 func NewKeyValueProperty(key, value string) (Property, error) {
+	if !validateKey(key) {
```

Comment:
```suggestion
	if !validateBaggageName(key) {
```

Reply:
Update the comment to include the same information as for `NewKeyValuePropertyRaw`

Reply:
same as https://github.com/open-telemetry/opentelemetry-go/pull/5132#discussion_r1688954419